lib/services/elasticsearch.js

```diff
@@ -919,429 +851,378 @@ class ElasticSearch extends Service {
+      esIndex = this._getESIndex(index, collection),
+      kuzzleMeta = {
         _kuzzle_info: {
-          active: true,
-          author: this._getUserId(request),
+          author: String(userId),
           updater: null,
           updatedAt: null,
-          deletedAt: null,
           createdAt: Date.now()
         }
-      });
+      },
+      {
+        rejected,
+        extractedDocuments,
+        documentsToGet
+      } = this._extractMDocuments(documents, kuzzleMeta, { prepareMGet: true });
 
     // prepare the mget request, but only for document having a specified id
-    let mgetRequest;
+    let mGetRequest;
 
-    if (extracted.mgetRequest.body.docs.length > 0) {
-      mgetRequest = this.client.mget(extracted.mgetRequest);
+    if (documentsToGet.length > 0) {
+      mGetRequest =
+        this.client.mget({ index: esIndex, body: { docs: documentsToGet }});
     } else {
-      mgetRequest = Bluebird.resolve({docs: []});
+      mGetRequest = Bluebird.resolve({ body: { docs: [] } });
     }
 
-    return mgetRequest
-      .then(mgetResult => {
+    return mGetRequest
+      .then(({ body }) => {
         const
-          esRequest = initESRequest(request,
-            ['consistency', 'refresh', 'timeout', 'fields']
-          ),
+          existingDocuments = body.docs,
+          esRequest = {
+            index: esIndex,
+            body: [],
+            refresh,
+            timeout
+          },
           toImport = [];
 
         let idx = 0;
 
-        esRequest.body = [];
-
-        for (let i = 0; i < extracted.documents.length; i++) {
-          const document = extracted.documents[i];
-
-          if (typeof document._id === 'string') {
-            // document._id should always be equal to mgetResult.docs[idx]._id
-            // if the document is not found in ES, or if it is in the trashcan,
-            // then we accept to create (or replace) it
-            if (!mgetResult.docs[idx].found ||
-              (mgetResult.docs[idx]._source._kuzzle_info &&
-                !mgetResult.docs[idx]._source._kuzzle_info.active)) {
-              esRequest.body.push(
-                {index:
-                  {_index: index,
-                    _type: type,
-                    _id: document._id}
-                });
-              esRequest.body.push(document._source);
-              toImport.push(document);
-            } else {
-              extracted.rejected.push(
-                {document: request.input.body.documents[i],
-                  reason: 'document already exists'}
-              );
-            }
+        for (let i = 0; i < extractedDocuments.length; i++) {
+          const document = extractedDocuments[i];
+
+          // Documents are retrieved in the same order than we got them from user
+          if (typeof document._id === 'string'
+            && existingDocuments[idx].found
+          ) {
+            rejected.push({
+              document: document,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use shorthand for property "document". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3498)